Formula/texlive.rb

```diff
@@ -0,0 +1,170 @@
+    end
+
+    resource("install-tl").stage do
+      cd "tlpkg" do
+        (libexec/"share/tlpkg").install "installer"
+      end
+    end
+
+    resource("texlive-texmf").stage do
+      (libexec/"share").install "texmf-dist"
+    end
+
+    # Clean unused files
+    rm_rf libexec/"share/texmf-dist/doc"
+    rm_rf libexec/"share/tlpkg/installer/wget"
+    rm_rf libexec/"share/tlpkg/installer/xz"
+
+    args = [
+      "--prefix=#{libexec}",
+      "--disable-debug",
+      "--disable-dependency-tracking",
+      "--disable-dvisvgm", # needs its own formula
+      "--disable-missing",
+      "--disable-native-texlive-build", # needed when doing a distro build
+      "--disable-ps2eps", # provided by ps2eps formula
+      "--disable-psutils", # provided by psutils formula
+      "--disable-static",
+      "--disable-t1utils", # provided by t1utils formula
+      "--enable-build-in-source-tree",
+      "--enable-shared",
+      "--enable-compiler-warnings=yes",
+      "--with-banner-add=/#{tap.user}",
+      "--with-system-clisp-runtime=system",
+      "--with-system-cairo",
+      "--with-system-freetype2",
+      "--with-system-gd",
+      "--with-system-gmp",
+      "--with-system-graphite2",
+      "--with-system-harfbuzz",
+      "--with-system-icu",
+      "--with-system-libpng",
+      "--with-system-mpfr",
+      "--with-system-ncurses",
+      "--with-system-pixman",
+      "--with-system-potrace",
+      "--with-system-zlib",
+    ]
+
+    args << if OS.mac?
+      "--without-x"
+    else
+      # Make sure xdvi uses xaw, even if motif is available
+      "--with-xdvi-x-toolkit=xaw"
+    end
+
+    system "./configure", *args
+    system "make"
+    system "make", "install"
+    system "make", "texlinks"
+
+    # Create tlmgr config file.  This file limits the actions that the user
+    # can perform in 'system' mode, which would write to the cellar.  'tlmgr' should
+    # be used with --usermode whenever possible.
+    (libexec/"share/texmf-config/tlmgr/config").write <<~EOS
+      allowed-actions=candidates,check,dump-tlpdb,help,info,list,print-platform,print-platform-info,search,show,version,init-usertree
+    EOS
+
+    # Create and link some directories into prefix and share.  Both links are needed
+    # because some scripts expect the directories to be in TEXMFROOT, and
+    # others in TEXMFROOT/share.
+    mkdir libexec/"share/texmf-var/tex/generic/config"
+    texdirs = %w[texmf-config texmf-dist texmf-var tlpkg]
+    texdirs.each do |texdir|
+      prefix.install_symlink libexec/"share"/texdir
+      share.install_symlink libexec/"share"/texdir
+    end
+
+    # Delete some binaries that are provided by existing formulae as newer versions.
+    rm libexec/"bin/latexindent" # provided by latexindent formula
+    rm libexec/"bin/latexdiff" # provided by latexdiff formula
+    rm libexec/"bin/latexdiff-vc" # provided by latexdiff formula
+    rm libexec/"bin/latexrevise" # provided by latexdiff formula
```

Comment:
We have a bunch of other `latex*` formulae (`latex2{html,rtf}`, `latexml`). Are there no conflicts from these formulae?

Reply:
I tested them and there were no conflicts.  So at the moment I think we've covered all the conflicts I'm aware of.